zebra-consensus/src/primitives/groth16.rs

```diff
@@ -0,0 +1,93 @@
+//! Async Groth16 batch verifier service
+
+use std::{
+    future::Future,
+    mem,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::future::{ready, Ready};
+use once_cell::sync::Lazy;
+use rand::thread_rng;
+use tokio::sync::broadcast::{channel, RecvError, Sender};
+use tower::Service;
+use tower_batch::{Batch, BatchControl};
+use tower_fallback::Fallback;
+use tower_util::ServiceFn;
+
+use bellman::{
+    groth16::{batch, PreparedVerifyingKey},
+    VerificationError,
+};
+use pairing::bls12_381::Bls12;
+
+/// A Groth16 verification item, used as the request type of the service.
+pub type Item = batch::Item<Bls12>;
```

Comment:
This prevents the `E: ... ` bounds from propagating through `zebra-consensus`.

Reply:
thank goodness 😅